tests/test-pkcs11-tool-test.sh

```diff
@@ -23,21 +23,39 @@ echo "======================================================="
 $PKCS11_TOOL --test -p "${PIN}" --module "${P11LIB}"
 assert $? "Failed running tests"
 
-echo "======================================================="
-echo "Test objects URI"
-echo "======================================================="
-$PKCS11_TOOL -O 2>/dev/null | grep 'uri:' 2>/dev/null >/dev/null
-assert $? "Failed running objects URI tests"
-$PKCS11_TOOL -O 2>/dev/null | grep 'uri:' | awk -F 'uri:' '{print $2}' | tr -d ' ' | grep ^"pkcs11:" 2>/dev/null >/dev/null
-assert $? "Failed running objects URI tests"
+# done with $P11LIB
 
-echo "======================================================="
-echo "Test slots URI"
-echo "======================================================="
-$PKCS11_TOOL -L 2>/dev/null | grep 'uri' 2>/dev/null >/dev/null
-assert $? "Failed running slots URI tests"
-$PKCS11_TOOL -O 2>/dev/null | grep 'uri' | awk -F 'uri*:' '{print $2}' | tr -d ' '  | grep ^"pkcs11:" 2>/dev/null >/dev/null
-assert $? "Failed running slots URI tests"
+# test if OpenSC can access a reader with a token using OpenSC module
+# to test how OpenSC creates 
+# softhsm2 module does not count 
+#
+OPENSC_TOOL="../src/tools/opensc-tool"
+if [[ -f "$OPENSC_TOOL" ]] ; then 
+    $OPENSC_TOOL -a
```

Comment:
>  It should test the URIs of the softhsm pkcs11 module 

But the original code: https://github.com/OpenSC/OpenSC/blob/master/tests/test-pkcs11-tool-test.sh#L29
does not have: "--module "${P11LIB}" so the default module would always fail when run under the the CI, as pcsc is not setup as softhsms module does not need it. 
 
The fact that the tests do not have the "--module "${P11LIB}" I interpreted as  deliberately left off to use the OpenSC module.  The tests would only run if there was a module that had a token. Without the tests would fail
so the `opensc-tool -a` would return 77 to skip of no token was available.     

The name of these tests are "Test objects URI"  and "Test slots URI" but the URI is generated in the pkcs11-tool, from info returned by the module so this "test"  looks more like a test of pkcs11-tool.c code then the module. 

Also https://github.com/OpenSC/OpenSC/blob/master/tests/Makefile.am#L58-L60 said these test where expected to fail and I move them out of that list, and made them fork, with either the  softhsm2 token, or a real token when I would run make check on my machine. 

So what are these tests really trying to do and why are they listed in  XFAIL_TESTS?

Reply:
That was probably overlook as the whole test was failing previously. If we change it to use softhsm as the rest of the tests and it will work, its preferred than skipping it.

Reply:
OK, removing using `opensc-tool -a ` and adding "--module "${P11LIB}" to URI tests.